packages/contracts/contracts/deploy/ControlledDAppFactory.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: MIT
+/// @title Controlled DApp Factory
+pragma solidity ^0.8.13;
+
+import {AccessControl} from "@openzeppelin/contracts/access/AccessControl.sol";
+import {IConsensus} from "@cartesi/rollups/contracts/consensus/IConsensus.sol";
+import {CartesiDApp} from "@cartesi/rollups/contracts/dapp/CartesiDApp.sol";
+import {ICartesiIPFSDAppFactory} from "../factory/ICartesiIPFSDAppFactory.sol";
+import {IFinancialProtocol} from "../protocol/IFinancialProtocol.sol";
+
+/// @notice This dapp factory implements the IFinancialProtocol with total control over the execution of the validator.
+/// @dev This can be used for test scenarios, where you can manuall start and stop the validator by sending a transaction.
+contract ControlledDAppFactory is
+    AccessControl,
+    ICartesiIPFSDAppFactory,
+    IFinancialProtocol
+{
+    ICartesiIPFSDAppFactory public immutable factory;
+
+    /// Create a new role identifier for the controller role, they can start/stop dapps
+    bytes32 public constant CONTROLLER_ROLE = keccak256("CONTROLLER_ROLE");
+
+    constructor(ICartesiIPFSDAppFactory _factory) {
+        factory = _factory;
+        _grantRole(CONTROLLER_ROLE, msg.sender);
+    }
+
+    /// @notice Instantiate a new dapp by delegating to a ICartesiIPFSDAppFactory, and emit an event requesting
+    /// validation without any financial incentives
+    /// @dev This function delegates the construction of the dapp to an IIPFSCartesiDAppFactory, and emits an event
+    /// requesting execution of validation until an "infinite" point in time in the future, which corresponds to saying
+    /// the validator should run with no financial incentives. It's a decision of the validator to accept this rule by
+    /// monitoring this contract for instantiation of new dapps.
+    /// @param _consensus The consensus to which the DApp should be subscribed
+    /// @param _dappOwner The address that should own the DApp
+    /// @param _templateHash The hash of the initial state of the Cartesi Machine
+    /// @param _cid The IPFS CID of the cartesi machine snapshot
+    /// @return The application
+    function newApplication(
+        IConsensus _consensus,
+        address _dappOwner,
+        bytes32 _templateHash,
+        string memory _cid
+    ) external override returns (CartesiDApp) {
+        // delegate call to factory, owner must be this contract, so we can register with payment system
```

Comment:
Once again, you haven't even registered the DApp in a payment system afterward. If this is the intended behavior, then we can pass `_dappOwner` directly to the call to `newApplication`, and not have to call `transferOwnership` later.